### PR TITLE
updated backend for stepper location

### LINKED ISSUE #92: BackEnd for stepper location
Implement BackEnd according to stepper location

### PR DESCRIPTION
**Pull request by SachkovaYuliia**
Before testing, please make sure that your .env file contains COUNTRYSTATECITY_API_KEY.

### CODE REVIEW

File: src/services/locationServiceCountries.js [hunk 1]
```diff
@@ -1,24 +1,54 @@
 const { createError } = require('~/utils/errorsHelper')
 const { COUNTRYSTATECITY_API_ISSUE } = require('~/consts/errors')
 
+const createRequestOptions = () => {
+  const headers = new Headers()
+  headers.append('X-CSCAPI-KEY', process.env.COUNTRYSTATECITY_API_KEY)
```
Comment (tufned) [inline src/services/locationServiceCountries.js]: you can move `process.env.COUNTRYSTATECITY_API_KEY` to constants

File: src/services/locationServiceCountries.js [hunk 2]
```diff
@@ -1,24 +1,54 @@
...
+
       const result = await response.json()
-      return result
+      const formattedCountriesData = formatCountriesResponse(result)
+      this.countriesCache = formattedCountriesData
+
+      return formattedCountriesData
     } catch (error) {
-      throw createError(401, COUNTRYSTATECITY_API_ISSUE)
+      throw createError(500, COUNTRYSTATECITY_API_ISSUE)
```
Comment (tufned) [inline src/services/locationServiceCountries.js]: it is better to change 500 status code to something else (for example to 404)

File: src/services/locationServiceCities.js [hunk 3]
```diff
@@ -1,24 +1,53 @@
...
       const result = await response.json()
-      return result 
+
+      const formattedCitiesData = formatCitiesResponse(result, countryName)
+      locationServiceCities.citiesCache.push({ iso2: iso2_code, cities: formattedCitiesData })
+
+      return formattedCitiesData
     } catch (error) {
-      throw createError(401, COUNTRYSTATECITY_API_ISSUE)
+      throw createError(500, COUNTRYSTATECITY_API_ISSUE)
```
Comment (tufned) [inline src/services/locationServiceCities.js]: it is better to use other status code instead of 500

File: src/initialization/serverSetup.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+require('dotenv').config()
```
Comment (tufned) [inline src/initialization/serverSetup.js]: do we need this line?

File: src/controllers/locationServiceCountries.js [hunk 5]
```diff
@@ -4,8 +4,12 @@ const {
 const locationServiceCountries = require('~/services/locationServiceCountries')
 
 const getCountryList = async (req, res) => {
-  const country_list = await locationServiceCountries.getCountries(API_KEY)
-  res.status(200).json(country_list)
+  try {
+    const country_list = await locationServiceCountries.getCountries(API_KEY)
+    res.status(200).json(country_list)
+  } catch (error) {
+    res.status(error.statusCode || 500).json({ error: error.message || 'Failed to fetch countries' })
```
Comment (tufned) [inline src/controllers/locationServiceCountries.js]: it's better to use other status code instead of 500

**Review by HalynaProlubshchykova — APPROVED** [review]

**Review by MrAK-dev — APPROVED** [review]

**Review by olehpona — APPROVED** [review]

**Review by tufned — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**